tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/polyparser/PolytreeParse.scala

```diff
@@ -347,30 +348,39 @@ object PolytreeParse {
   private def fromConllHelper(
     fileLines: Iterator[String],
     useGoldPosTags: Boolean
-  ): Stream[PolytreeParse] = {
-
-    val (nextParseLines, rest) = fileLines.span(_.nonEmpty)
-    constructFromConllString(nextParseLines, useGoldPosTags) #:: (if (rest.hasNext) {
-      rest.next()
-      if (rest.hasNext) {
-        // check for more in case last line in file is newline
-        fromConllHelper(rest, useGoldPosTags)
-      } else {
-        Stream.empty
+  ): Iterator[PolytreeParse] = {
+
+    // Groups the lines of the file by parse (parses are identified via a blank line
+    // in the file).
+    def groupby[T](iter: Iterator[T])(startsGroup: T => Boolean): Iterator[Iterator[T]] =
```

Comment:
Speed. It was really slow for long files, for some reason. My theory is that my streaming logic was somehow done incorrectly (though I don't honestly know why the other implementation was so slow).